waku/waku_archive/archive.nim

```diff
@@ -111,7 +111,7 @@ proc handleMessage*(w: WakuArchive,
       msgReceivedTime = if msg.timestamp > 0: msg.timestamp
                         else: getNanosecondTime(getTime().toUnixFloat())
 
-    trace "handling message", pubsubTopic=pubsubTopic, contentTopic=msg.contentTopic, timestamp=msg.timestamp, digest=msgDigest, messageHash=msgHash
+    trace "handling message", pubsubTopic=pubsubTopic, contentTopic=msg.contentTopic, timestamp=msg.timestamp, digest=toHex(msgDigest.data), messageHash=toHex(msgHash)
```

Comment:
Upon latest ligging research, can we just move any conversion/calculation away from logging line?
Until we don't know what cause double call of any proc appears in the log arguments.
It is also involves memory allocation for the hex string, I would avoid that.

Reply:
But this leads to another problem, if trace is disabled the hash would still be calculated which would reduce performance even in normal scenario. 
Unless we can access and use the logging macro and put the hash calculation statement under it.

Reply:
I actually just discussed this point with Zoltán :))

In such cases we can check the log level and compute depending on it. I just added this fix without doing a specific check for log level, as I'm doing an extremely cheap computation - converting a constant size hash from bytes to hex.

For bigger computations, I guess the best thing is to compute it based on the log level.

Most important thing might be to figure out why computation happens twice when added as a log argument. The biggest problem here is the double memory allocation.

Reply:
Yeah exactly. Agreed to enter an issue for chronicles lib. That can solve this issue in an optimal way.
Within this in mind, I would agree to keep it if you think so. I would have like to claim the attention on hidden costs of such calls.

Reply:
That's a good approach, let's check it directly with Chronicles. In the meantime, I'll leave the computation outside the log. It's a super simple/cheap one, so if we can avoid having them inside then better